src/plugins/intel_cpu/src/scratch_pad.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "common/memory.hpp"
+#include <memory>
+#include "cpu_memory.h"
+#include "dnnl_extension_utils.h"
+
+namespace ov {
+namespace intel_cpu {
+
+class ScratchPad {
+    dnnl::engine eng;
+    Memory m_curScratchpadMem;
+    size_t m_curScratchpadSize = 0;
+
+public:
+    ScratchPad(dnnl::engine eng) : eng(eng), m_curScratchpadMem(eng) {}
+
+    void setScratchPad(std::unordered_map<int, dnnl::memory> & args, const dnnl::memory::desc & md) {
+        // Scratch pad memory only increase for now
+        // Memory re-allocation is free from multi-threading safety issue
+        // as long as it's called from stream's scheduling thread
+        auto requiredSize = md.get_size();
+        if (m_curScratchpadSize < requiredSize) {
```

Comment:
note here I could use `m_curScratchpadMem.redefineDesc(DnnlExtensionUtils::makeDescriptor(md));` unconditionally, but I found that `redefineDesc()` itself have many overhead before it decide to not resize existing memory, since this function is called in performance critical path, I decided to add a condition here as a short cut

Reply:
Got it. But might there be situations when we have the same size but different descriptors for the Scratchpad?

Reply:
Oh, you are right, also it's possible that the required md is of smaller size, in that case we still need to call `redefineDesc()` to get a correct memory (in `GetPrimitive()`) with exact md as required.  so I guess we cannot skip the call to `redefineDesc()` and all overhead inside that function.  you can see this is exactly why I didn't choose Memory class in the original design, Memory class, although provided many useful function, but also introduced a lot of overhead for example the internal conversion from `dnnl::memory::desc` to `DnnlMemoryDescPtr`, which maybe useful when you implementing some kernel, but here this kind of overhead is not needed at all.

Reply:
It only reveals that conversion from dnnl::memory::desc tp DnnlBlockedDesc is not effective enough. This is because we planned to add lazy evaluations to the class wrapper around   dnnl::memory::desc, but something went wrong. If this is not a blocked descriptor, then we have nearly zero overhead. All the other stages are pretty much the same as they were in the original design, just checking the size and creation of a new memory object. But, going this way we use a power of concealing the complexity. However, if the overheads have a not acceptable level, we should consider change the proposed design.